manager/src/main/java/com/redhat/service/bridge/manager/workers/Worker.java

```diff
@@ -0,0 +1,46 @@
+package com.redhat.service.bridge.manager.workers;
+
+import com.redhat.service.bridge.infra.models.dto.ManagedResourceStatus;
+import com.redhat.service.bridge.manager.models.ManagedResource;
+import com.redhat.service.bridge.manager.models.Work;
+
+/**
+ * Handles completion of {@link Work} for a {@link ManagedResource}.
+ * 
+ * If {@link ManagedResource#getStatus()} is {@link ManagedResourceStatus#ACCEPTED} it is assumed the {@link ManagedResource} needs to
+ * be created and its dependencies created by {@link Worker#createDependencies(Work, ManagedResource)}.
+ * 
+ * If the {@link ManagedResource#getStatus()} is {@link ManagedResourceStatus#DEPROVISION} it is assumed the {@link ManagedResource} needs to
+ * be destroyed and its dependencies deleted by {@link Worker#deleteDependencies(Work, ManagedResource)}.
+ * 
+ * Other values of {@link ManagedResourceStatus} when the {@link Work} is {@link WorkManager#schedule(ManagedResource)} will result
+ * in the {@link Work} remaining incomplete indefinitely.
+ * 
+ * @param <T>
+ */
+public interface Worker<T extends ManagedResource> {
+
+    /**
+     * Execute work. When complete {@link WorkManager#complete(Work)} should be invoked.
+     * 
+     * @param work
+     * @return true if the Work was processed, otherwise false.
+     */
+    boolean handleWork(Work work);
```

Comment:
@manstis I couldn't immediately determine the need to return a `boolean` from this method as I don't see anything making use of it.

Reply:
Oeeee. You'll like this.

The return value is not really needed however there is an issue with (possibly) Quarkus and VertX's `EventBus#requestAndForget(String, Object)`. If a value is not returned the logs show occurrences of "ERROR: Failed to send reply.(TIMEOUT,-1) Timed out after waiting 30000(ms)". See https://stackoverflow.com/questions/49449257/vertx-timeout-in-message-reply.

So, I thought it "nicer" to return "something" (other than `null`) to avoid the errors.

Reply:
I've added a comment to the code explaining it.

Reply:
I can put together a test for the Quarkus team.